### PR TITLE
[Deprecation] Re-add ORKTextAnswerFormat and ORKRegistrationStep regex methods

### PR DESCRIPTION
**Pull request by rsanchezsaez**
This PR re-adds the previously removed `ORKTextAnswerFormat` and `ORKRegistrationStep` regex methods, and marks them as deprecated. Fixes https://github.com/ResearchKit/ResearchKit/issues/968.

I added comments explaining that they will be deprecated in *v1.5.0* (the next stable release), and that they're scheduled for removal in *v1.6.0*. I assume that following a policy of removing deprecated methods after one stable release is ok. @syoung-smallwisdom: Let me know if you disagree.

Also, I moved every deprecated method to the new `ORKDeprecated.h`/`.m` files, and I added specific deprecation attributes and comments. All previously deprecated methods are also scheduled for removal in *v1.6.0*.

### CODE REVIEW

File: ResearchKit/1.4.1/ResearchKit.podspec [hunk 1]
```diff
@@ -1,15 +0,0 @@
-Pod::Spec.new do |s|
```
Comment (syoung-smallwisdom) [inline ResearchKit/1.4.1/ResearchKit.podspec]: Is removing this file intensional? I don't use cocoa-pods and don't really remember.

Comment (syoung-smallwisdom) [inline ResearchKit/1.4.1/ResearchKit.podspec]: This doesn't seem to be related. Why are you removing this?

File: ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m [hunk 2]
```diff
@@ -52,20 +52,6 @@ @interface ORKHealthQuantityTypeRecorder () {
 
 @end
 
-#ifdef __IPHONE_10_0
```
Comment (syoung-smallwisdom) [inline ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m]: This was only needed during last summer so folks could develop with either Xcode 8 beta or Xcode 7. I've removed it in a branch that is waiting for acceptance of my other pull requests.

File: ResearchKit/Common/ORKOrderedTask.h [hunk 3]
```diff
@@ -390,17 +390,6 @@ ORK_EXTERN ORKTrailMakingTypeIdentifier const ORKTrailMakingTypeIdentifierB;
...
- */
-+ (ORKOrderedTask *)audioTaskWithIdentifier:(NSString *)identifier
-                     intendedUseDescription:(nullable NSString *)intendedUseDescription
-                          speechInstruction:(nullable NSString *)speechInstruction
-                     shortSpeechInstruction:(nullable NSString *)shortSpeechInstruction
-                                   duration:(NSTimeInterval)duration
-                          recordingSettings:(nullable NSDictionary *)recordingSettings
-                                    options:(ORKPredefinedTaskOption)options __deprecated;
-
-/**
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKOrderedTask.h]: This deprecation warning is removed in https://github.com/ResearchKit/ResearchKit/pull/959 and will cause a merge conflict with that branch. (It's already in stable 1.4 and 1.5 will be released soon.)

File: ResearchKit/Common/ORKOrderedTask.h [hunk 4]
```diff
@@ -427,13 +416,6 @@ ORK_EXTERN ORKTrailMakingTypeIdentifier const ORKTrailMakingTypeIdentifierB;
                                                       duration:(NSTimeInterval)duration
                                                    handOptions:(ORKPredefinedTaskHandOption)handOptions
                                                        options:(ORKPredefinedTaskOption)options;
-/**
- @Deprecated
- */
-+ (ORKOrderedTask *)twoFingerTappingIntervalTaskWithIdentifier:(NSString *)identifier
-                                        intendedUseDescription:(nullable NSString *)intendedUseDescription
-                                                      duration:(NSTimeInterval)duration
-                                                       options:(ORKPredefinedTaskOption)options __deprecated;
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKOrderedTask.h]: This is handled in https://github.com/ResearchKit/ResearchKit/pull/959 and will cause a merge conflict with that branch.

File: ResearchKit/Common/ORKOrderedTask.m [hunk 5]
```diff
@@ -356,17 +356,6 @@ void ORKStepArrayAddStep(NSMutableArray *array, ORKStep *step) {
...
-+ (ORKOrderedTask *)twoFingerTappingIntervalTaskWithIdentifier:(NSString *)identifier
-                                       intendedUseDescription:(NSString *)intendedUseDescription
-                                                     duration:(NSTimeInterval)duration
-                                                      options:(ORKPredefinedTaskOption)options {
-    return [self twoFingerTappingIntervalTaskWithIdentifier:identifier
-                                     intendedUseDescription:intendedUseDescription
-                                                   duration:duration
-                                                handOptions:0
-                                                    options:options];
-}
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKOrderedTask.m]: https://github.com/ResearchKit/ResearchKit/pull/959

File: ResearchKit/Common/ORKOrderedTask.m [hunk 6]
```diff
@@ -519,24 +508,6 @@ + (ORKOrderedTask *)twoFingerTappingIntervalTaskWithIdentifier:(NSString *)ident
...
-    return [self audioTaskWithIdentifier:identifier
-                  intendedUseDescription:intendedUseDescription
-                       speechInstruction:speechInstruction
-                  shortSpeechInstruction:shortSpeechInstruction
-                                duration:duration
-                       recordingSettings:recordingSettings
-                         checkAudioLevel:NO
-                                 options:options];
-}
-
```
Comment (syoung-smallwisdom) [inline ResearchKit/Common/ORKOrderedTask.m]: https://github.com/ResearchKit/ResearchKit/pull/959

File: ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m (hunk 2 above)
Comment (syoung-smallwisdom) [inline ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m]: I submitted stripping this as a separate PR. https://github.com/ResearchKit/ResearchKit/pull/970

File: ResearchKit/1.4.1/ResearchKit.podspec (hunk 1 above)
Comment (rsanchezsaez) [inline ResearchKit/1.4.1/ResearchKit.podspec]: Yeah, this is intentional. It seems like an unused duplicate of `ResearchKit.podspec` on the root folder.

File: ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m (hunk 2 above)
Comment (rsanchezsaez) [inline ResearchKit/ActiveTasks/ORKHealthQuantityTypeRecorder.m]: Done.

File: ResearchKit/Common/ORKOrderedTask.h (hunk 3 above)
Comment (rsanchezsaez) [inline ResearchKit/Common/ORKOrderedTask.h]: Done.

File: ResearchKit/Common/ORKOrderedTask.h (hunk 4 above)
Comment (rsanchezsaez) [inline ResearchKit/Common/ORKOrderedTask.h]: Done.

File: ResearchKit/Common/ORKOrderedTask.m (hunk 5 above)
Comment (rsanchezsaez) [inline ResearchKit/Common/ORKOrderedTask.m]: Done.

File: ResearchKit/Common/ORKOrderedTask.m (hunk 6 above)
Comment (rsanchezsaez) [inline ResearchKit/Common/ORKOrderedTask.m]: Done.

**Comment by rsanchezsaez** [comment]
I restored the unrelated deprecated methods to avoid *merge conflicts*. ;)
